ff/src/fields/models/fp/montgomery_backend.rs

```diff
@@ -241,12 +240,10 @@ pub trait MontConfig<const N: usize>: 'static + Sync + Send + Sized {
             r.b1[i] = carry;
             carry = 0;
         }
-        r.b1[N - 1] >>= 63;
-        for i in 0..N {
-            r[2 * (N - 1) - i] = (r[2 * (N - 1) - i] << 1) | (r[2 * (N - 1) - (i + 1)] >> 63);
-        }
-        for i in 3..N {
-            r[N + 1 - i] = (r[N + 1 - i] << 1) | (r[N - i] >> 63);
+
+        r.b1[N - 1] = r.b1[N - 2] >> 63;
+        for i in 2..(2 * N - 1) {
+            r[2 * N - i] = (r[2 * N - i] << 1) | (r[2 * N - (i + 1)] >> 63);
```

Comment:
```suggestion
        for i in 1..(2 * N) {
            r[2 * N - i] = (r[2 * N - i] << 1) | (r[2 * N - (i + 1)] >> 63);
```

Reply:
Is line 244 outside of the loop for performace reasons?
Otherwise we could remove the explicit cases of r[2*N-1] and r[1] before and after the loop and instead do:

```
for i in 1..(2 * N) {
            r[2 * N - i] = (r[2 * N - i] << 1) | (r[2 * N - (i + 1)] >> 63);
```

Reply:
I ran some benchmarks: https://gist.github.com/mmagician/ee80f60e402a9e29ad8dff0a62424edd. As expected the result is pretty inconclusive, given how small the change would be.

Reply:
Yeah, those were outside the loop for perf reasons. Let me try your version, also comparing the performance against master.

Reply:
Hm I can detect a small but non-zero perf difference between the master and your version (the diff b/w my version and master is smaller).